src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -22,6 +23,17 @@ public static class RoutingEndpointConventionBuilderExtensions
     /// An empty collection means any host will be accepted.
     /// </param>
     /// <returns>A reference to this instance after the operation has completed.</returns>
+    /// <remarks>
+    /// APIs that rely on the <see href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Host">Host header</see>, such as
+    /// <see cref="HttpRequest.Host"/> and <see cref="RequireHost"/>, are vulnerable to potential spoofing by clients.
+    ///
+    /// To prevent host and port spoofing, use one of the following approaches:
+    ///
+    /// <list type="bullet">
+    ///   <item>Use <see cref="HttpContext.Connection"/> (<see cref="ConnectionInfo.LocalPort"/>) where the ports are checked.</item>
+    ///   <item>Employ <see href="https://learn.microsoft.com/aspnet/core/fundamentals/servers/kestrel/host-filtering">Host filtering</see>.</item>
+    /// </list>
```

Comment:
Host filtering is not a mitigation for spoofing. All it does is check the Host header.

https://github.com/dotnet/aspnetcore/blob/79ef5e329b1e31c3775a1977798253cc8f7da6cc/src/Middleware/HostFiltering/src/HostFilteringMiddleware.cs#L84

```suggestion
    /// To prevent host and port spoofing, use <see href="https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.connections.features.itlshandshakefeature.hostname">ITlsHandshakeFeature.HostName</see> to verify the server name used during the TLS handshake, and <see cref="HttpContext.Connection"/> (<see cref="ConnectionInfo.LocalPort"/>) to verify the local port the connection was accepted on.
```

I think trusting requests only if the arrive on specific local ports is a pretty rare scenario, but I know @drewnoakes was working on exactly this for Aspire. It's definitely true you should not rely on the host header for this if you do not trust the client or proxy.